docs/DeveloperGuide.md

```diff
@@ -1,38 +1,575 @@
+- **`FileManager#createDirIfNotExist()`** and **`FileManager#createFileIfNotExist()`**: Ensure that the necessary directory and file exist before saving.
+
+#### Storage converts JSON data into Java objects:
+- **`Storage#loadProgrammeList()`**: Converts the `ProgrammeList` JSON data into a `ProgrammeList` object using `programmeListFromJson()`.
+- **`Storage#loadHistory()`**: Converts the `History` JSON data into a `History` object using `historyFromJson()`.
+- **`Storage#saveData()`**: Converts `ProgrammeList` and `History` into JSON using `createJSON()`, and passes it to `FileManager#save()`.
+
+#### DateSerializer is responsible for:
+- **`DateSerializer#serialize()`**: Converts `LocalDate` into `JsonElement`.
+- **`DateSerializer#deserialize()`**: Converts `JsonElement` back into `LocalDate`.
+
+These operations are exposed in the `Model` interface, allowing seamless saving and loading of `ProgrammeList` and `History` data.
+
+
+### Flow of Operations
+Given below is an example usage scenario and how the save/load mechanism behaves at each step.
+
+**Step 1.** The user launches the application for the first time. A `Storage` object is initialized by `BuffyBuddy`, and it attempts to load data from
+the file using `FileManager`. If no data file exists, `Storage` initializes an empty `ProgrammeList` and `History`.
+
+**Step 2.** The user interacts with the application by adding programmes or logging workout activities and meals, modifying both the
+`ProgrammeList` and `History`. These changes are stored temporarily in memory, but no data is saved to the file at this point.
+
+**Step 3.** When the user chooses to exit the application, `Model#saveData()` is triggered, which in turn calls `Storage#saveData()`.
+At this point, `Storage` converts the current `ProgrammeList` and `History` into JSON format using the `createJSON()` method and passes
+the `JsonObject` to `FileManager#save()`.
+
+**Step 4.** The `FileManager` saves the updated `JsonObject` to the data file, ensuring that the user's changes are preserved for the
+next session. If necessary, `FileManager#createDirIfNotExist()` and `FileManager#createFileIfNotExist()` ensure that the correct directories
+and files are in place before saving.
+
+**Step 5.** The next time the user launches the application, `Storage#loadProgrammeList()` and `Storage#loadHistory()` are called, which
+load the data from the file via `FileManager#load()`. The loaded data is then converted from JSON back into `ProgrammeList` and `History`
+objects, restoring the user's previous session.
+
+The following sequence diagram shows how a load operation for ProgrammeList goes through the Storage component:
+![Sequence Diagram for Load operation](./images/LoadProgrammeList_Seq_Dia.jpg)
+
+The following sequence diagram shows how a save operation goes through the Storage component:
+![Sequence Diagram for Save operation](./images/Save_Seq-Dia.jpg)
```

Comment:
For both addition of programmeListJsonObject and historyJsonObject using the add() function, you could be more specific and mention it as the arguments of the function.

```suggestion
add("programmeList", programmeListToJson(programmeList))
```

Since the add method depends on the Json object, you should consider adding it.
Since the add function depends on the Json object